### PR TITLE
try catch for data processing so that the service isn't killed 

### PR DESCRIPTION
**Pull request by jh-bate**
@darinkrauss I was thinking as simple as this?

### CODE REVIEW

File: lib/jellyfishService.js [hunk 1]
```diff
@@ -339,80 +339,84 @@ function createServer(serverConfig, mongoClient, seagullClient, userApiClient, g
...
             }
-          } else {
-            res.status(200).send(duplicates);
           }
-        }
-      );
+        );
+      } catch (ex){
+        log.warn(ex, 'an uncaught error occurred!?');
+        res.status(500).send('an uncaught error occurred');
```
Comment (darinkrauss) [inline lib/jellyfishService.js]: I'm not sure how the client will handle this, so you might want to check with Jana first, but shouldn't we send back an expected JSON error object?  Something like `{statusCode: 500, message: 'Error getting user private data'}`?

File: lib/jellyfishService.js [hunk 2]
```diff
@@ -339,80 +339,84 @@ function createServer(serverConfig, mongoClient, seagullClient, userApiClient, g
...
-              res.send(500);
+              res.status(200).send(duplicates);
             }
-          } else {
-            res.status(200).send(duplicates);
           }
-        }
-      );
+        );
+      } catch (ex){
```
Comment (darinkrauss) [inline lib/jellyfishService.js]: Missed a space between the closing paren and opening brace.

**Comment by darinkrauss** [comment]
@jh-bate This is what I was thinking.  Have you tested it to ensure it works as expected?  Also, we'll want to resolve the one comment I made about the JSON error response before merging.

**Comment by jh-bate** [comment]
@darinkrauss yeah testing at the moment along with the other JF issue

**Pull request closed without merging**